platform/microprofile/src/main/java/org/trellisldp/microprofile/WebApplication.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.trellisldp.microprofile;
+
+import static java.util.Arrays.asList;
+import static java.util.stream.Collectors.joining;
+import static java.util.stream.Collectors.toSet;
+import static org.slf4j.LoggerFactory.getLogger;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.util.Set;
+
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.ws.rs.ApplicationPath;
+import javax.ws.rs.core.Application;
+
+import org.slf4j.Logger;
+import org.trellisldp.auth.oauth.OAuthFilter;
+import org.trellisldp.http.*;
+
+/**
+ * Web Application wrapper.
+ */
+@ApplicationPath("/")
+@ApplicationScoped
+public class WebApplication extends Application {
+
+    private static final Logger LOGGER = getLogger(WebApplication.class);
+
+    @Inject
+    private TrellisHttpResource httpResource;
+
+    @Inject
+    private AgentAuthorizationFilter agentFilter;
+
+    @Inject
+    private TrellisHttpFilter httpFilter;
+
+    @Inject
+    private CrossOriginResourceSharingFilter corsFilter;
+
+    @Inject
+    private CacheControlFilter cacheFilter;
+
+    @Inject
+    private OAuthFilter oauthFilter;
+
+    @PostConstruct
+    private void init() {
+        printBanner("Trellis Microprofile");
+    }
+
+    @Override
+    public Set<Object> getSingletons() {
+        return asList(httpResource, httpFilter, corsFilter, cacheFilter, agentFilter, oauthFilter).stream()
+            .collect(toSet());
+    }
+
+    private void printBanner(final String name) {
+        final URL resource = Thread.currentThread().getContextClassLoader().getResource("banner.txt");
```

Comment:
This seems a bit fragile to me. Can we not inline the banner in this class? If not, just how big is it?

Reply:
I have the `banner.txt` resource in the `trellis-app` artifact, which gets used by both the dropwizard and µprofile artifacts. The code here is pretty much just a simplification of how DropWizard prints out the banner.

Reply:
I'm not going to lie down in the road over this. I've just seen too many troubles come from loading resources from the "current thread" classpath to feel super comfortable, so I'd rather see it inlined, but until we actually have a problem, whatevs.

Reply:
I've updated this slightly to (hopefully) make this less fragile.